services-custom/s3-transfermanager/src/main/java/software/amazon/awssdk/custom/s3/transfer/MultipartUploadConfiguration.java

```diff
@@ -111,6 +114,33 @@ public static MultipartUploadConfiguration defaultConfig() {
         return DEFAULT;
     }
 
+    private static Boolean resolveEnableMultipartUploads(Boolean configured) {
+        return Optional.ofNullable(configured).orElse(DEFAULT_ENABLE_MULTIPART_UPLOAD);
+    }
+
+    private static Long resolveMultipartUploadThreshold(Long configured) {
+        return Optional.ofNullable(configured).orElse(DEFAULT_MULTIPART_UPLOAD_THRESHOLD);
+    }
+
+    private static Integer resolveMaxUploadPartCount(Integer configured) {
+        return Optional.ofNullable(configured).orElse(DEFAULT_MAX_UPLOAD_PART_COUNT);
+    }
+
+    private static Long resolveMinUploadPartSize(Long configured) {
+        return Optional.ofNullable(configured).orElse(DEFAULT_MIN_UPLOAD_PART_SIZE);
+    }
+
+    private void validateConfig() {
+        Validate.isPositive(multipartUploadThreshold, "multipartUploadThreshold");
+        Validate.isPositive(minUploadPartSize, "minUploadPartSize");
+        Validate.isTrue(minUploadPartSize <= multipartUploadThreshold,
```

Comment:
why `minUploadPartSize` should be less than `multipartUploadThreshold`? It makes sense but should not be a requirement. Customer might have large objects like 1 GB and they want part size to be 50 MB or so. In that we should do 50MB parts instead of forcing them to 16MB parts.

Reply:
Makes sense. I'll change to 5MiB, since that's the absolute min set by S3.